pkg/platform/runtime/buildscript/buildscript.go

```diff
@@ -62,44 +68,59 @@ func NewScript(data []byte) (*Script, error) {
 		return nil, errs.Wrap(err, "Could not parse build script")
 	}
 
+	// Construct the equivalent buildexpression.
+	bytes, err := json.Marshal(script)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not marshal build script to build expression")
+	}
+	expr, err := buildexpression.New(bytes)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not construct build expression")
+	}
+	script.Expr = expr
+
 	return script, nil
 }
 
+func NewScriptFromBuildExpression(expr *buildexpression.BuildExpression) (*Script, error) {
+	return &Script{Expr: expr}, nil
+}
```

Comment:
Shouldn't this also create the Script? Is there a potential that a user calling a script method could get an error because the struct only has an expression assigned?

Reply:
Not yet. Any merges that need to take place will be done before writing to disk. That's how it was before, so I don't believe I've changed anything.